src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -2540,6 +2558,10 @@ new_fstatat(int dirfd, const char *__restrict path, struct stat *__restrict stat
 			return new_xstat(1, path, stat_buf);
 	}
 
+	if (dirfd >= FD_FILE_BASE && dirfd < FD_DIR_BASE && path[0] == 0)
+		/* same as fstat for a file. May need further work to handle flags */
+		return fstat(dirfd, stat_buf);
+
```

Comment:
Is this related to the commit message?  Same applies for the rest of the code changes in this file.

Reply:
This is a corner case of fstatat(). dirfd is allowed to be a fd of a regular file.
Yes. There are probably a few places where we need to handle flags. I added a comment here to make sure I will not forget it in future.